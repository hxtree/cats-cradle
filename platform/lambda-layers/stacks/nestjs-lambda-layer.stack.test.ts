import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { NestJsLambdaLayerStack } from './nestjs-lambda-layer.stack';

describe('MainStack', () => {
  it('should match snapshot test', () => {
    const app = new cdk.App();
    const stack = new NestJsLambdaLayerStack(app, 'MyTestStack');
    const template = Template.fromStack(stack);
    expect(template).toMatchSnapshot();
  });
});
