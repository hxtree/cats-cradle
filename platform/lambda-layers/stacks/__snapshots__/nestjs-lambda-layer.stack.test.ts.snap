// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainStack should match snapshot test 1`] = `
{
  "Outputs": {
    "NestJSLayerVersionArn": {
      "Value": {
        "Ref": "NestJsAppLayerF9ACDB87",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "NestJsAppLayerF9ACDB87": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "CompatibleArchitectures": [
          "x86_64",
          "arm64",
        ],
        "Content": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "2640a05f1ab7f3b50acbdb9b4601513df930c8841534d0432bf112cb1b9f61e5.zip",
        },
        "LayerName": {
          "Fn::Join": [
            "",
            [
              "arn:aws:lambda:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":layer:NestJsAppLayer",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::LayerVersion",
      "UpdateReplacePolicy": "Delete",
    },
    "lambdalayernestjslatestversionB188CF78": {
      "Properties": {
        "Description": "NestJS Lambda Layer Latest Version",
        "Name": "lambda-layer-nestjs-latest-version",
        "Type": "String",
        "Value": {
          "Ref": "NestJsAppLayerF9ACDB87",
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
